com.reprezen.swagedit/src/com/reprezen/swagedit/model/NodeDeserializer.java

```diff
@@ -0,0 +1,84 @@
+package com.reprezen.swagedit.model;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+public class NodeDeserializer extends JsonDeserializer<AbstractNode> {
+
+    @Override
+    public AbstractNode deserialize(JsonParser p, DeserializationContext context)
+            throws IOException, JsonProcessingException {
+
+        final Model model = (Model) context.getAttribute("model");
+        final AbstractNode parent = (AbstractNode) context.getAttribute("parent");
+        final JsonPointer ptr = (JsonPointer) context.getAttribute("pointer");
+
+        JsonLocation startLocation = p.getTokenLocation();
+        if (p.getCurrentToken() == JsonToken.FIELD_NAME) {
+            p.nextToken();
+        }
+
+        if (p.getCurrentToken() == JsonToken.START_OBJECT) {
+
+            ObjectNode node = new ObjectNode(parent, ptr, p.getCurrentLocation());
+            node.setStartLocation(startLocation);
+            model.add(node);
+
+            while (p.nextToken() != JsonToken.END_OBJECT) {
+                String name = p.getCurrentName();
+
+                JsonPointer pp = JsonPointer.compile(ptr.toString() + "/" + name.replaceAll("/", "~1"));
+
+                context.setAttribute("parent", node);
+                context.setAttribute("pointer", pp);
+
+                AbstractNode v = deserialize(p, context);
+                v.setProperty(name);
+                node.put(name, v);
+            }
+
+            node.setEndLocation(p.getCurrentLocation());
+            return node;
+
+        } else if (p.getCurrentToken() == JsonToken.START_ARRAY) {
```

Comment:
What about extracting methods to handle `START_OBJECT`,  `START_ARRAY ` and the default case instead of keeping them all inside a 70-line long method?